src/Parallel/Printf/Printf.hpp

```diff
@@ -118,12 +133,26 @@ inline void printf(const std::string& format, Args&&... args) {
  */
 template <typename... Args>
 inline void printf_error(const std::string& format, Args&&... args) {
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#pragma GCC diagnostic ignored "-Wformat-security"
-  detail::call_printer(
-      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
-      [](auto... a) { CkError(a...); }, format, std::forward<Args>(args)...);
-#pragma GCC diagnostic pop
+  detail::print_message(true, format, std::forward<Args>(args)...);
 }
+
+/// Chare outputting all Parallel::printf results.
+class PrinterChare : public CBase_PrinterChare {
+ public:
+  PrinterChare() = default;
+  explicit PrinterChare(CkMigrateMessage* /*msg*/) {}
+
+  /// Prints a message to stdout or stderr.
+  static void print(bool error, const std::vector<char>& message);
```

Comment:
We could probably add an `fprint` that also takes a `std::string` as an argument that then prints to a text file. This will be useful for logging

Reply:
If it's very simple to add, then I'd say do it. But if it's more than a few lines then maybe do it in a follow up?